### DISCUSSION TITLE
Error 400 on POST when i tried to use reactive search on a cluster hosted in elasti.co

### DISCUSSION THREAD

**Discussion by Kumkum29** [appbaseio/reactivesearch#2303] [General]
Hello,

I followed this post to use reactive search with a cluster hosted on elasti.co: https://docs.reactivesearch.io/docs/reactivesearch/react/v3/overview/quickstart/

I set CORS directives in elasticsearch.yml ans set the default ReactiveBase like this : 
`    <ReactiveBase
      url="https://xxxxxxx.elastic-cloud.com"
      app="my_index"
      credentials="username:password"
      enableAppbase={false}
    >`

But, when i load the page, and the search engine (only a searchbox for the moment), i get this error : 

`POST https://xxxxxxx.aws.elastic-cloud.com/my_index/_reactivesearch 400 (Bad Request)`

This is normal, the "/_reactivesearch" method does not exist by default on elasticsearch...

Can you tell me how to fix this problem?

Thanks for your help!!

**Comment by Kumkum29** [comment]
I see this post : https://github.com/appbaseio/reactivesearch/discussions/1933, . This seems to be the same problem, but there is no answer.

**Comment by Kumkum29** [comment]
If I use a port in the 'url' prop, like this ':9200', i get no error on page load, but an error when the search are launched...

**Comment by siddharthlatest** [comment]
I would suggest using the reactivesearch-api server. You can run this with Docker in this way: https://github.com/appbaseio/reactivesearch-api-docker. The ReactiveSearch v4 library expects to connect to the reactivesearch-api server.

**Reply by Kumkum29** [reply]
Hello Siddharthlatest !

Thanks for your feedback. I'll look into using reactivesearch_api.

I tried using version 3, and I have the same problem. The searchbox makes a call to /_reactivesearch (get an error 400), a method that does not exist natively in elasticsearch.

Should I use a proxy to forward the call to /_reactivesearch ? Or should I configure my cluster in another way ? The cluster is hosted on elasti.co.

If I use the default cluster, shown in the documentation, 'https://appbase-demo-ansible-abxiydt-arc.searchbase.io' I have no problem.

I used reactivesearch a few years ago, and everything worked right away with a cluster on elasti.co. I'm a bit lost...

Your advice and help is invaluable in resolving the problem.

Thanks.

**Reply by Kumkum29** [reply]
@Siddharthlatest,

I decrypted the code where the '_reactivesearch' method is located. Here it is: 

`            function b(e, t, n) {
                var r = a(t),
                    i = o(e);
                if (!0 !== i) throw i;
                var s = { settings: r, query: e };
                return (
                    this.mongodb && Object.assign(s, { mongodb: l(this.app, this.mongodb) }),
                    this.performFetchRequest({ method: "POST", path: "_reactivesearch", body: s, headers: u(this.enableTelemetry, !this.mongodb), isRSAPI: !0, isMongoRequest: !!this.mongodb, params: n })
                );
            }`

Can you explain to me the usefulness of this additional method? For the stats ? Is there a solution to not use this method (in app settings) ?
